src/core/mesh/qgsmeshlayerrenderer.cpp

```diff
@@ -55,9 +55,29 @@ QgsMeshLayerRenderer::QgsMeshLayerRenderer( QgsMeshLayer *layer, QgsRenderContex
   copyScalarDatasetValues( layer );
   copyVectorDatasetValues( layer );
 
+  assignDefaultScalarShader();
+
   calculateOutputSize();
 }
 
+void QgsMeshLayerRenderer::assignDefaultScalarShader( )
+{
+  if ( mScalarDatasetValues.isEmpty() || mRendererScalarSettings.isEnabled() )
+    return; // no need for default shader, either rendering is off or we already have some shader
+
+  double vMin = *std::min_element( mScalarDatasetValues.constBegin(), mScalarDatasetValues.constEnd() );
+  double vMax = *std::max_element( mScalarDatasetValues.constBegin(), mScalarDatasetValues.constEnd() );
+
+  QList<QgsColorRampShader::ColorRampItem> lst;
+  lst << QgsColorRampShader::ColorRampItem( vMin, Qt::blue, QString::number( vMin ) );
+  lst << QgsColorRampShader::ColorRampItem( vMax, Qt::red, QString::number( vMax ) );
```

Comment:
shall we use the color ramp from settings? 
```
    QgsSettings settings;
    QString defaultPalette = settings.value( QStringLiteral( "/Raster/defaultPalette" ), "Spectral" ).toString();
    btnColorRamp->setColorRampFromName( defaultPalette );
```
The blue-red color ramp is quite ugly as a default :-)